### ISSUE TITLE
Update attributes names from official classic TBC client "leak"

### ISSUE THREAD

**Issue by T1ti** [stoneharry/WoW-Spell-Editor#139]
The 2.5 TBC Client leaked the attributes names blizzard uses publicly, it has more precise names and would replace some "unknowns". For example attribute 44 was guessed as "pickpocket" because they use it, but for blizzard it is actually "failure can break stealth"

https://wow.tools/dbc/?dbc=enumeratedstring&build=2.5.1.39640#page=5

It's TBC so it might not entirely apply 1:1 but most of it seems to match

**Screenshots**

**Additional context**

**Comment by stoneharry** [comment]
I started working on this locally a while ago but it's not exactly a fun task manually updating all the attributes... anyone can do this, it's all loaded from the language files. 😉 

For example, updated attributes:
```
<!--#region some array of strings-->
    <system:String x:Key="attFlags_strings">
		Proc Failure Burns Charge |
		Uses Ranged Slot |
		On Next Swing (No Damage) |
		Do Not Log Immune Misses |
		Is Ability |
		Is Trade Skill |
		Passive |
		Do Not Display (Spellbook, Aura Icon, Combat Log) |
		Do Not Log |
		Held Item Only |
		On Next Swing |
		Wearer Casts Proc Trigger |
		Server Only |
		Allow Item Spell In PvP |
		Indoors Only |
		Outdoors Only |
		Not Shapeshifted |
		Only Stealthed |
		Do Not Sheath |
		Spell Damage scales with Level |
		Cancels Auto Attack Combat |
		Impossible to Dodge, Parry or Block |
		Track Target while Casting |
		Castable While Dead |
		Castable While Mounted |
		Start Cooldown after Aura Fades |
		Negative, Aura is Debuff |
		Castable While Sitting |
		Cannot be used in Combat |
		Unaffected by Invulnerability |
		Breakable by Damage |
		Aura Cannot be Cancelled
    </system:String>
```

I'll try to finish it off at some point soon if nobody else gets round to it.

**Comment by Chromie-WoW** [comment]
missed one

**Comment by stoneharry** [comment]
> missed one and renaming this wont matter if its not coded to do what it says right?

I missed this question - it's just a display in the program, renaming it in the languages file will not make any difference.

The order is important because the order determines the flag that is set.

Which one was missed?

**Comment by stoneharry** [comment]
These are my proposed string updates:
```xml
<system:String x:Key="attFlags_strings">
		Proc Failure Burns Charge |
		Uses Ranged Slot |
		On Next Swing (No Damage) |
		Do Not Log Immune Misses |
		Is Ability |
		Is Trade Skill |
		Passive |
		Do Not Display (Spellbook, Aura Icon, Combat Log) |
		Do Not Log |
		Held Item Only |
		On Next Swing |
		Wearer Casts Proc Trigger |
		Server Only |
		Allow Item Spell In PvP |
		Indoors Only |
		Outdoors Only |
		Not Shapeshifted |
		Only Stealthed |
		Do Not Sheath |
		Spell Damage scales with Level |
		Cancels Auto Attack Combat |
		Impossible to Dodge, Parry or Block |
		Track Target while Casting |
		Castable While Dead |
		Castable While Mounted |
		Start Cooldown after Aura Fades |
		Negative, Aura is Debuff |
		Castable While Sitting |
		Cannot be used in Combat |
		Unaffected by Invulnerability |
		Breakable by Damage |
		Aura Cannot be Cancelled
    </system:String>
    <system:String x:Key="attFlagsEx_strings">
        Dismiss Pet First |
        Drains All Power |
        Is Channelled |
        Cannot be Redirected |
        No Skill Increase |
        Allow While Stealthed |
        Is Self Channelled |
        Cannot be Reflected |
        Cannot Target in Combat |
        Initiates Combat (Enables Auto-Attack) |
        Generates No Threat |
        Aura Unique |
        Failure Breaks Stealth |
        Far Sight |
        Track Target while Channel;ing |
        Remove Auras on Immunity |
        Immunity to Hostile and Friendly Effects |
        No AutoCast (AI) |
        Prevents Animation |
        Cannot Target Self |
        Finishing Move - Damage |
        Threat only on miss |
        Finishing Move - Duration |
        Ignore Owner's Death |
        Special Skillup |
        Aura Stays After Combat |
        Require All Targets |
        Discount Power On Miss |
        Hidden in Aura Bar |
        Name in Channel Bar |
        Enable Spell when Dodged / Combo on Block |
        Cast When Learned
    </system:String>
    <system:String x:Key="attFlagsEx1_strings">
        Can Target Dead Unit or Corpse |
        No Shapeshift UI |
        Ignore Line of Sight |
        Allow Low Level Buff |
        Display in Stance Bar / Use Shapeshift Bar |
        Auto Repeat |
        Requires Untapped Target |
        Do Not Report Spell Failure |
        Include In Advanced Combat Log |
        Always Cast As Unit |
        Special Taming Flag |
        No Target Per-Second Costs |
        Chain From Caster |
        Preserve Enchant in Arena |
        Allow While Invisible |
        Do Not Consume if Gained During Cast |
        No Active Pets |
        Do Not Reset Combat Timers |
        Requires Dead Pet / No Jump While Cast Pending |
        Don't Need Shapeshift |
        Initiate Combat Post-Cast (Enables Auto-Attack) |
        Fail on all targets immune |
        No Initial Threat |
        Proc Cooldown On Failure |
        Item Cast With Owner Skill |
        Don't Block Mana Regen |
        No School Immunities |
        Ignore Weapon Skill |
        Not an Action |
        Cannot Crit |
        Active Threat |
        Retain Item Cast
    </system:String>
    <system:String x:Key="attFlagsEx2_strings">
        PvP Enabling |
        No Proc Equip Requirement |
        No Casting Bar Text |
        Blockable Spell / Completely Blocked |
        Ignore Resurrection Timer |
        No Durability Loss |
        No Avoidance |
        Stack for Different Casters |
        Only Target Players |
        Not a Proc |
        Requires Main-Hand |
        Battleground Only |
        Only Target Ghosts |
        Hide Channel Bar |
        Hide In Raid Filter |
        Normal Ranged Attack (Auto-Shoot) |
        Suppress Caster Procs |
        Suppress Target Procs |
        Cannot Miss |
        Disable Procs / Instant Target Procs |
        Death Persistent |
        Only Proc Outdoors |
        Casting Cancels Autorepeat (Mainline: Do Not Trigger Target Stand) |
        No Damage History |
        Requires Off-Hand |
        Can Proc From Procs |
        Only Proc on Caster |
        Ignore Caster and Target Restrictions |
        Ignore Caster Modifiers |
        Do not Display Range |
        Not On AOE Immune
    </system:String>
    <system:String x:Key="attFlagsEx3_strings">
        Ignore All Resistances |
        Proc Only on Caster |
        Continue to Tick while Offline |
        No Helpful Threat |
        No Harmful Threat |
        Allow Client Targeting |
        Cannot Be Stolen |
        Allow Cast While Casting / Triggered |
        Fixed Damage / Ignore Damage Taken Modifiers |
        Activate from Event / Combat Feedback When Usable |
        Spell vs Extended Cost / Weapon Speed Cost Scaling |
        No Partial Immunity |
        Aura Is Buff |
        Do Not Log Caster |
        Damage doesn't Break Auras / Reactive Damage Proc |
        Not In Spellbook |
        Not Usable in Arena |
        Usable in Arena |
        Bouncy Chain Missiles |
        Allow Proc While Sitting |
        Don't Check Selfcast Power / Aura Never Bounces |
        Allow Entering Arena |
        Proc Suppress Swing Anim |
        Suppress Weapon Procs |
        Auto Ranged Combat |
        Pet Scaling / Owner Power Scaling |
        Can only be used in Flying Areas |
        Force Display Castbar |
        Ignore Combat Timer |
       	Aura Bounce Fails Spell |
        Obsolete |
        Use Facing From Spell
    </system:String>
    <system:String x:Key="attFlagsEx4_strings">
        Allow Actions During Channel |
        No Reagent Cost With Aura |
        Remove Entering Arena |
        Allow While Stunned |
        Triggers Channeling |
        Single-Target Spell / Limit N |
        Ignore Area Effect PvP Check |
        Not On Player |
        Not On Player Controlled NPC |
        Start Periodic at Aura Apply / Extra Initial Period |
        Do Not Display Duration |
        Allow Target of Target as Target / Implied Targeting |
        Melee Chain Targeting (Cleave) |
        Spell Haste Affects Periodic (Duration) |
        Not Available While Charmed |
        Treat as Area Effect (Inflict on Multiple Targets) |
        Aura Affects Not Just Req. Equipped Item |
        Allow While Fleeing (Feared) |
        Allow While Confused |
        AI Doesn't Face Target / Don't Turn during Casting |
        Do Not Attempt a Pet Resummon When Dismounting |
        Ignore Target Requirements |
        Not On Trivial |
        No Partial Resists |
        Ignore Caster Requirements |
        Always Line of Sight |
        Always AOE Line of Sight |
        No Caster Aura Icon |
        No Target Aura Icon |
      	Aura Unique Per Caster |
        Always Show Ground Texture |
        Add Melee Hit Rating / AoE Taunt
    </system:String>
    <system:String x:Key="attFlagsEx5_strings">
        No Cooldown On Tooltip |
        Do Not Reset Cooldown In Arena |
        Not an Attack / Ignore Caster Auras |
        Can Assist Immune PC / Assist Ignore Immune Flag |
        Ignore For Mod Time Rate |
        Do Not Consume Resources |
        Floating Combat Text On Cast |
        Aura Is Weapon Proc |
        Do Not Chain To Crowd-Controlled Targets |
        Allow On Charmed Targets |
        No Aura Log |
        Not In Raid Instances |
        Allow While Riding Vehicle |
        Can Target Invisible / Ignore Phase Shift |
        AI Primary Ranged Attack |
        No Pushback |
        No Jump Pathing |
        Allow Equip While Casting |
        Originate From Controller |
        Delay Combat Timer During Cast |
        Aura Icon Only For Caster (Limit 10) |
        Show Mechanic as Combat Text |
        Absorb Cannot Be Ignore |
        Taps immediately |
        Can Target Untargetable |
        Doesn't Reset Swing Timer if Instant |
        Vehicle Immunity Category |
        Ignore Healing Modifiers |
        Do Not Auto Select Target with Initiates Combat |
        Ignore Caster Damage Modifiers |
        Disable Tied Effect Points |
        No Category Cooldown Mods
    </system:String>
    <system:String x:Key="attFlagsEx6_strings">
        Allow Spell Reflection / Feign Death |
        No Target Duration Mod |
        Disable Aura While Dead |
        Cheat Spell / Debug Spell |
        Treat as Raid Buff |
        Can Be Multi Cast (Totem) |
        Don't Cause Spell Pushback |
        Prepare for Vehicle Control End |
        Horde Only |
        Alliance Only |
        Dispel Removes Charges |
        Can Cause Interrupt (Interrupt only Non-Player) |
        Can Cause Silence |
        No UI Not Interruptible |
        Recast On Resummon (Raise Dead) |
        Reset Swing Timer at spell start |
        Only In Spellbook Until Learned (Restore Secondary Power) |
        Do Not Log PvP Kill |
        Attack on Charge to Unit |
        Report Spell failure to unit target (Zone Teleport) |
        No Client Fail While Stunned, Fleeing, Confused (Blink, Divine Shield, Ice Block) |
        Retain Cooldown Through Load |
        Ignores Cold Weather Flying Requirement |
        No Attack Dodge |
        No Attack Parry |
        No Attack Miss |
        Treat as NPC AoE |
        Bypass No Resurrect Aura |
        Do Not Count For PvP Scoreboard / Consolidated Raid Buff |
        Reflection Only Defends |
        Can Proc From Suppressed Target Procs |
        Always Cast Log / Client Indicator
    </system:String>
```

**Comment by stoneharry** [comment]
Closing this because it is included in Release Build v1.9.9, thanks all.
